src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -20,15 +20,37 @@ internal abstract partial class ConversionsBase
 
         internal readonly bool IncludeNullability;
 
-        protected ConversionsBase(AssemblySymbol corLibrary, int currentRecursionDepth, bool includeNullability)
+        private ConversionsBase _lazyOtherNullability;
+
+        protected ConversionsBase(AssemblySymbol corLibrary, int currentRecursionDepth, bool includeNullability, ConversionsBase otherNullabilityOpt)
         {
             Debug.Assert((object)corLibrary != null);
+            Debug.Assert(otherNullabilityOpt == null || includeNullability != otherNullabilityOpt.IncludeNullability);
+            Debug.Assert(otherNullabilityOpt == null || currentRecursionDepth == otherNullabilityOpt.currentRecursionDepth);
 
             this.corLibrary = corLibrary;
             this.currentRecursionDepth = currentRecursionDepth;
             IncludeNullability = includeNullability;
+            _lazyOtherNullability = otherNullabilityOpt;
         }
 
+        internal ConversionsBase WithNullability(bool includeNullability)
```

Comment:
>WithNullability [](start = 33, length = 15)

doc on this would be useful too, since there is some trickery to minimize allocations